### PR TITLE
use GroupVersion in APIGroupVersion for api installer

### PR DESCRIPTION
**Pull request by deads2k**
Re-open of https://github.com/kubernetes/kubernetes/pull/17175

@caesarxuchao @liggitt

### CODE REVIEW

File: pkg/apiserver/apiserver_test.go [hunk 1]
```diff
@@ -57,10 +57,19 @@ func convert(obj runtime.Object) (runtime.Object, error) {
 }
 
 // This creates fake API versions, similar to api/latest.go.
-const testVersion = "version"
-const newVersion = "version2"
-
-var versions = []string{testVersion, newVersion}
+var testAPIGroup = "test.group"
+var testGroupVersion = unversioned.GroupVersion{Group: testAPIGroup, Version: "version"}
+var testVersion = testGroupVersion.String()
```
Comment (caesarxuchao) [inline pkg/apiserver/apiserver_test.go]: Can we remove the testVersion? It's containing group/version.

**Comment by deads2k** [comment]
@caesarxuchao Sorry about the mistaken merge before.  This is the same pull, plus @liggitt's additional tests.

File: pkg/apiserver/watch_test.go [hunk 2]
```diff
@@ -178,8 +178,8 @@ func TestWatchParamParsing(t *testing.T) {
 
 	dest, _ := url.Parse(server.URL)
 
-	rootPath := "/api/" + testVersion + "/watch/simples"
-	namespacedPath := "/api/" + testVersion + "/watch/namespaces/other/simpleroots"
+	rootPath := "/" + prefix + "/" + testVersion + "/watch/simples"
```
Comment (caesarxuchao) [inline pkg/apiserver/watch_test.go]: You will catch this once testVersion is deleted.

File: pkg/apiserver/apiserver_test.go (hunk 1 above)
Comment (deads2k) [inline pkg/apiserver/apiserver_test.go]: > Can we remove the testVersion? It's containing group/version.

This intersects with the `RESTMapping` that I'm touching now.  May I do it as part of that pull?  The use of `version` in and out of the `RESTMapper` was inconsistent between tests and code

File: pkg/apiserver/watch_test.go (hunk 2 above)
Comment (deads2k) [inline pkg/apiserver/watch_test.go]: > You will catch this once testVersion is deleted.

Ok, point taken.  I will update.

File: pkg/apiserver/apiserver_test.go (hunk 1 above)
Comment (caesarxuchao) [inline pkg/apiserver/apiserver_test.go]: Sure. Thanks.

File: pkg/apiserver/watch_test.go (hunk 2 above)
Comment (caesarxuchao) [inline pkg/apiserver/watch_test.go]: I'm ok if you want to update this in the RESTMapper pull as well.

**Comment by caesarxuchao** [comment]
I just took another pass closely. LGTM. Thank you very much @deads2k.

**Comment by deads2k** [comment]
@k8s-bot test this please

**Comment by liggitt** [comment]
@k8s-bot test this

**Comment by caesarxuchao** [comment]
@k8s-bot test this please

**Comment by piosz** [comment]
@k8s-bot e2e test this please

**Comment by deads2k** [comment]
clean rebase, re-added lgtm

**Comment by caesarxuchao** [comment]
ref #17216

**Pull request merged**